dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/transform/strategy/WebAssetStrategy.java

```diff
@@ -0,0 +1,41 @@
+package com.dotmarketing.portlets.contentlet.transform.strategy;
+
+import static com.dotmarketing.portlets.htmlpageasset.business.HTMLPageAssetAPI.URL_FIELD;
+
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.liferay.portal.model.User;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * This a super class intended to model anything that can reside on the we and therefore have a url
+ * It it intended to calculate the url associated to the identifier and avoid caching issues.
+ * @param <T>
+ */
+public abstract class WebAssetStrategy <T extends Contentlet>  extends
+        AbstractTransformStrategy<T> {
+
+    WebAssetStrategy(final TransformToolbox toolBox) {
+        super(toolBox);
+    }
+
+    public final void apply(final T contentlet, final Map<String, Object> map,
+            final Set<TransformOptions> options, User user) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'user' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5596402)